app/charts/chart-config-ui-options.ts

```diff
@@ -91,6 +86,46 @@ export type EncodingOption =
       field: "useAbbreviations";
     };
 
+const makeOnColorComponentChange = (type: "areaLayer" | "symbolLayer") => {
+  return (
+    draft: ConfiguratorStateConfiguringChart,
+    action: {
+      type: "scaleType";
+      value: "continuous" | "discrete";
+    }
+  ) => {
```

Comment:
Since the sideEffect is receiving the field, couldn't we just get the field type at runtime ?

```
const colorComponentChange = (draft, action, field) => {
  const type = field.type
....
}
```